Library/Homebrew/extend/pathname.rb

```diff
@@ -271,7 +271,7 @@ def uninstall_info
   end
 
   # Writes an exec script in this folder for each target pathname.
-  sig { params(targets: T::Array[Pathname]).void }
+  sig { params(targets: T.any(T::Array[Pathname], Pathname)).void }
```

Comment:
```suggestion
  sig { params(targets: T.any(T::Array[T.any(String, Pathname)], String, Pathname)).void }
```

We also allow `String`s here (mostly interpolated ones like `#{bin}/foo`), or arrays of `String`s.

https://github.com/Homebrew/brew/blob/75aee5ec5ccb7cebc05086fde14a640452c44e38/Library/Homebrew/extend/pathname.rb#L283

There is [substantial usage outside of Homebrew](https://github.com/search?q=NOT%20repo%3AHomebrew%2Fhomebrew-core%20NOT%20is%3Afork%20%22write_exec_script%20%5C%22%22&type=code) which we don't want to break, so let's just update the type signature here.